assembly/compliance-tool/src/main/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/reporter/ReporterUtils.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Bosch.IO GmbH 2020.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.frontend.compliancetool.sw360.reporter;
+
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360Release;
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360SparseRelease;
+
+import java.util.Collection;
+
+class ReporterUtils {
+    private ReporterUtils() {}
+
+    static String[] printCollectionOfReleases(Collection<SW360Release> releases) {
+        return releases.stream()
+                .map(ReporterUtils::releaseCsvPrintRow)
+                .toArray(String[]::new);
+    }
+
+    static String releaseCsvPrintHeader() {
+        return "release id;name;version;" +
+                "coordinates;" +
+                "main license ids;" +
+                "overridden license;declared license;observed license;" +
+                "copyrights;clearing state;change status;" +
+                "download url; release tag url; software heritage id";
+    }
+
+    private static String releaseCsvPrintRow(SW360Release release) {
+        return String.format("%s;%s;%s;%s;%s;%s;%s;%s;%s;%s;%s;%s;%s;%s",
+                release.getId(), release.getName(), release.getVersion(),
+                collectionToString(release.getCoordinates().values()),
+                collectionToString(release.getMainLicenseIds()),
+                release.getOverriddenLicense(), release.getDeclaredLicense(), release.getObservedLicense(),
+                release.getCopyrights(), release.getClearingState(), release.getChangeStatus(),
+                release.getDownloadurl(), release.getReleaseTagUrl(), release.getSoftwareHeritageId());
+    }
```

Comment:
The csv printer only takes complete records ( meaning rows). If I would get the fields as Object[] I would have to iterate over the arrays of fields and give them individually as records, making the chance high to have mismatched placements. 
Did I understand your suggestion correctly?

Reply:
I was referring to the printRecord() method of CSVPrinter which expects an Object[] (as var args) as argument. It can be passed an object of fields and prints them as a single row. This method is already used currently, but only a single string (with the already generated row) is passed in. The generation of this string is done manually, and I was hoping that it could be delegated to CSVPrinter.